migrations/lib/converters/discourse/steps/category_custom_fields.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Migrations::Converters::Discourse
+  class CategoryCustomFields < ::Migrations::Converters::Base::ProgressStep
+    attr_accessor :source_db
+
+    def max_progress
+      @source_db.count <<~SQL
+        SELECT COUNT(*) FROM category_custom_fields
+      SQL
+    end
+
+    def items
+      @source_db.query <<~SQL
+        SELECT category_id, name, value
+        FROM category_custom_fields
+      SQL
+    end
+
+    def process_item(item)
+      IntermediateDB::CategoryCustomField.create(
```

Comment:
`existing_id` support might be needed here, Chat seeds category custom fields for `SiteSetting.staff_category_id` and `SiteSetting.general_category_id`.  Confirm if it's needed

Reply:
The importer should do the mapping for existing categories. We shouldn't need to worry about in other steps I think.

Reply:
Could you explain how envisage this working?